keter.cabal

```diff
@@ -64,6 +64,7 @@ library
       random >=1.2.1 && <1.3,
       regex-tdfa >=1.3.1 && <1.4,
       stm >=2.5.0 && <2.6,
+      tagged >= 0.1.0 && <1.0,
```

Comment:
About this — I'd deliberately picked the version range to the widest sanely possible.

As far as using just the names `Tagged(Tagged), untag` — those are provided in every version of that package, even historic tagged-0.1.0 circa 2010.

This is to ensure the widest compatibility window.